src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -667,12 +662,10 @@ protected ExecutableCodeBinder TryGetBodyBinderFromSyntax(BinderFactory binderFa
         /// which might return locations of partial methods.
         /// </summary>
         public override ImmutableArray<Location> Locations
-        {
-            get
-            {
-                return this.locations;
-            }
-        }
+            => ImmutableArray.Create(_location);
```

Comment:
Worried this is now silently allocating on every call where it used to not. Did you consider `[Obsolete]` this method and point people to `Location` porperty?

Reply:
great idea.  i will.

to your concern, i haven't seen the calls through Locations be problematic in practice.  it's very rare for that to actually happen afaict.  within the compiler we basically don't do it (we only often fetch the first location for diagnostic purposes).

Reply:
Ugh.  I wish that worked.  Unfortunately:

![image](https://user-images.githubusercontent.com/4564579/232134611-2f4ed272-9f27-46be-a2c9-b397c0598248.png)

Note that hte user of Locations in the bottom code doesn't warn.  Looks like obsolete is checked for the top virtual in an override chain :'(

Reply:
I've seen _many_ cases where `Location[]` instances were problematic in a trace. This was one of the ones that wasn't problematic before. I didn't keep track of them because the problem was showing up from other types that didn't have the caches.

Reply:
> I've seen many cases where Location[] instances were problematic in a trace. 

These were addressed by: https://github.com/dotnet/roslyn/pull/67678

The core issue was that the compiler would often request the total array of locations just to always grab the first one for diagnostic purposes.  The recent PRs introduced a fast, non-allocating, path for all those cases to just grab a single Location to report.  

With this change (and a few mopups based on traces) we've gotten the compiler away from generically asking for the Locations array nearly everywhere.  Jared and i are also conversing about ways to stamp out its use entirely.